File: Jump_Search/Jump_search.py
# Python implementation of Jump search algorithm

from math import floor,sqrt

def jumpSearch(lst, num):
    n = len (lst)
    
    # calculate block size to jump
    step = floor(sqrt(n))
    prev = 0
    
    while lst[min(step, n)-1] < num:
        prev = step
        step += floor(sqrt(n))
        if prev >= n: return -1 # num not present in list
    # linear search through block for num
    while lst[prev] < num:
        prev += 1;
        if prev == min(step, n): return -1 # num not present in list
    if lst[prev] == num: return prev #num is present at position prev
    
    return -1

if __name__ == "__main__":
    print("Enter your number list: ")
    lst = list(map(int, input().split()))
    print("Enter number to search: ")
    num = int(input())
    res = jumpSearch(lst, num);
    if (res < 0): print ("{} is not present in your list".format(num))
    else: print ("{} is present at position {}".format(num, res))
    

File: Jump_Search/jumpSearch.js
function jumpSearch(arr, target) {

	let len = arr.length

	//Finding the block size for the jump
	let step = Math.floor(Math.sqrt(len));

	//blockStart is the starting index of the block that our target will be in
	let blockStart = 0, currentStep = step;

	while (arr[Math.min(currentStep, len) - 1] < target)
	{
		//as long as we haven't found the block, we move to the next block and check again
		blockStart = currentStep;
		currentStep += step;

		//If the next block is pass the length of the array, target doesn't exist, return -1
		if (blockStart >= len)
			return -1;
	}

	//Linear Search within the block until we find the possible index for the target
	while (arr[blockStart] < target)
	{
		blockStart++;

		//If we reached the next block or end of array, the target doesn't exist
		if (blockStart == Math.min(currentStep, len))
			return -1;
	}

	//Check if the element is the target, if not, target doesn't exist.
	if (arr[blockStart] == target)
		return blockStart
	else
		return -1;
}